mino/minogrpc/example_test.go

```diff
@@ -11,6 +14,110 @@ import (
 	"go.dedis.ch/dela/serde"
 )
 
+func TestRpcStreamTree(t *testing.T) {
+	NB_NODES := 200
+
+	nodes := createNodes(NB_NODES, false)
+	rpcs := createRpcs(nodes)
+	exchangeCertificates(nodes)
+	players := createPlayers(nodes)
+	msgs := generateMessages(NB_NODES)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for i, r := range rpcs {
+		go func(i int, rpc mino.RPC) {
+			sender, receiver, err := rpc.Stream(ctx, players)
+			if err != nil {
+				panic("stream failed: " + err.Error())
+			}
+
+			for _, n := range nodes {
+				addr := n.GetAddress()
+
+				msgSent := msgs[i]
+				err = <-sender.Send(exampleMessage{value: msgSent}, addr)
+				if err != nil {
+					panic("failed to send: " + err.Error())
+				}
+
+				from, msgReceived, err := receiver.Recv(ctx)
+				if err != nil {
+					panic("failed to receive: " + err.Error())
+				}
+
+				require.True(t, from.Equal(addr))
+				require.Equal(t, msgReceived, msgSent)
+			}
+		}(i, r)
+	}
+}
+
+func generateMessages(nbNodes int) []string {
+	var messages []string
```

Comment:
A good habit is to initialize the slice at the right size if you know it in advance. This is a simple memory optimization:

```go
messages := make([]string, nbNodes)

for i := 0; i < nbNodes; i++ {
  messages[i] = fmt.Sprintf("MSG%d", i)
}

return messages
```

same bellow